### DISCUSSION TITLE
rtsp-simple-server randomly  failing to launch with ERR: too many open files

### DISCUSSION THREAD

**Discussion by jav123456** [bluenviron/mediamtx#497] [Questions]
Hi,

I am using rtsp-simple-server on Centos 7.4. Sometimes it launches fine, but other times I get an error "ERR: too many open files". 

Example:
2021/08/12 12:09:43 I [0/0] rtsp-simple-server v0.16.0
2021/08/12 12:09:43 I [0/0] [RTSP] UDP/RTP listener opened on :8000
2021/08/12 12:09:43 I [0/0] [RTSP] UDP/RTCP listener opened on :8001
2021/08/12 12:09:43 I [0/0] [RTSP] TCP listener opened on :8554
2021/08/12 12:09:43 I [0/0] [RTMP] listener opened on :1935
2021/08/12 12:09:43 I [0/0] [HLS] listener opened on :8888
2021/08/12 12:09:43 I [0/0] ERR: too many open files

I also tested with v0.17, it has the same problem.

I have checked the number of opened files in the OS and it is fine, very inferior to the maximum.

When it happens, one thing that seems to help is stopping port forwarding the 8554 port (normally I have this port forwarded to other system) and then launch it  again before restarting port forwarding. But this workaround is not very  consistent, the problem seems to be a bit random in nature.

Does anybody knows what this error means and how to fix it?

**Comment by aler9** [comment]
The error is caused by the fact that your OS has a limit on the number of opened file descriptors. You can edit this limit by following a tutorial:

https://www.cyberciti.biz/faq/linux-increase-the-maximum-number-of-open-files/

**Reply by jav123456** [reply]
That was the first thing I checked, but it is not that. I only have 9000 opened , with  a 1 million limit.

[a comment by an automated account is not shown]
